### PR TITLE
[WebKit] Add support for Xcode 14 beta 1, 2 & 3

### PR DESCRIPTION
**Pull request by SotoiGhost**

### CODE REVIEW

File: src/wkwebkit.cs
Comment (dalexsoto) [inline src/wkwebkit.cs:1051]: Why is this commented out? It should either added or removed

File: src/wkwebkit.cs
Comment (dalexsoto) [inline src/wkwebkit.cs:1037]: Smells bad, macOS 13 but iOS 15,5 on a new member? I think they should be bumped to iOS 16

```suggestion
		[Mac (13,0), iOS (16,0), MacCatalyst (16,0), NoWatch, NoTV]
```

File: src/wkwebkit.cs
Comment (dalexsoto) [inline src/wkwebkit.cs:1033]: Why this is changed? I mean just double checking that this was intentional

File: src/wkwebkit.cs
Comment (dalexsoto) [inline src/wkwebkit.cs:1041]: Smells bad, macOS 13 but iOS 15,5 on a new member? I think they should be bumped to iOS 16

```suggestion
		[Mac (13,0), iOS (16,0), MacCatalyst (16,0), NoWatch, NoTV]
```

File: src/wkwebkit.cs
Comment (dalexsoto) [inline src/wkwebkit.cs:1045]: same smells bad, macOS 13 but iOS 15,5 on a new member? I think they should be bumped to iOS 16

```suggestion
		[Mac (13,0), iOS (16,0), MacCatalyst (16,0), NoWatch, NoTV]
```

File: src/wkwebkit.cs
Comment (dalexsoto) [inline src/wkwebkit.cs:1053]: Please make a separated issue about this if you haven't already and link it [here](https://github.com/xamarin/xamarin-macios/issues/15232) so we do not forget

**Review by dalexsoto — CHANGES_REQUESTED** [review]
Looking good!

File: src/wkwebkit.cs
Comment (SotoiGhost) [reply src/wkwebkit.cs:1045]: https://github.com/xamarin/xamarin-macios/pull/15525#discussion_r922506330

**Comment by SotoiGhost** [comment]
These methods depend on new API on UIKit. Already created an issue and added the issue [here](https://github.com/xamarin/xamarin-macios/issues/15232#issue-1265763007).

https://github.com/xamarin/xamarin-macios/blob/e05ad13a2392a5a1995b3941335ca9f1b3bce78f/src/wkwebkit.cs#L1049-L1055

File: src/wkwebkit.cs
Comment (mandel-macaque) [inline src/wkwebkit.cs:40]: Why the bump?

File: src/wkwebkit.cs
Comment (mandel-macaque) [reply src/wkwebkit.cs:1037]: I'd go for the largest number, since moving to an earlier one is simpler, the other way around is not.

**Review by mandel-macaque — APPROVED** [review]
+1 but I'd like answers before landing.

File: src/wkwebkit.cs
Comment (SotoiGhost) [reply src/wkwebkit.cs:40]: Apple's thought

```objc
// Xcode 14 beta 3
typedef NS_ENUM(NSInteger, WKFullscreenState) {
    WKFullscreenStateNotInFullscreen,
    WKFullscreenStateEnteringFullscreen,
    WKFullscreenStateInFullscreen,
    WKFullscreenStateExitingFullscreen,
} NS_SWIFT_NAME(WKWebView.FullscreenState) API_AVAILABLE(macos(13.0), ios(16.0));
```

**Review by dalexsoto — APPROVED** [review]

File: src/wkwebkit.cs
Comment (mandel-macaque) [inline src/wkwebkit.cs:1270]: Intro says that te property does NOT work on iOS.

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

**Comment by mandel-macaque** [comment]
@SotoiGhost the intro tests failures must be fixed:
```
FAIL] Selector not found for WebKit.WKWebpagePreferences : isLockdownModeEnabled in Boolean get_LockdownModeEnabled() on WebKit.WKWebpagePreferences 
[FAIL] Selector not found for WebKit.WKWebpagePreferences : setLockdownModeEnabled: in Void set_LockdownModeEnabled(Boolean) on WebKit.WKWebpagePreferences 
```

**Comment by mandel-macaque** [comment]
@SotoiGhost please check the intro tests, they seem to be valid.

**Comment by SotoiGhost** [comment]
PR already rebased with `xcode14` b4 bump. Waiting for CI results

**Comment by mandel-macaque** [comment]
xtro failures are fixed here: https://github.com/xamarin/xamarin-macios/pull/15625
intro is known: https://github.com/xamarin/maccore/issues/581
cecil tests are fixed here: https://github.com/xamarin/xamarin-macios/pull/15610

**Pull request merged**
